src/plots/plots.js

```diff
@@ -1387,8 +1387,8 @@ plots.plotAutoSize = function plotAutoSize(gd, layout, fullLayout) {
         // but don't enforce any ratio restrictions
         var computedStyle = isPlotDiv ? window.getComputedStyle(gd) : {};
 
-        newWidth = parseFloat(computedStyle.width) || fullLayout.width;
-        newHeight = parseFloat(computedStyle.height) || fullLayout.height;
+        newWidth = parseFloat(computedStyle.width) || parseFloat(computedStyle.maxWidth) || fullLayout.width;
+        newHeight = parseFloat(computedStyle.height) || parseFloat(computedStyle.maxHeight) || fullLayout.height;
```

Comment:
I'm curious. Can we have a situation where `computedStyle.height` > `computedStyle.maxHeight`?

If so, here we honour `computedStyle.height`, but shouldn't `computedStyle.maxHeight` win out? 

(This is probably more a of theoretical question than a real-life situation).